parse.go

```diff
@@ -322,6 +322,8 @@ func setSlice(dest reflect.Value, values []string) error {
 		elem = elem.Elem()
 	}
 
+	dest = reflect.MakeSlice(reflect.SliceOf(elem), 0, len(values))
```

Comment:
Simplest thing here may be just 

```go
if !dest.IsNil() {
  dest.SetLen(0)
}
````

That should truncate the length of the existing slice to zero without allocating a new slice. If dest refers to a `nil` value then `reflect.Append` below will allocate a new empty slice.